webapp/channels/src/components/new_search/search_box.tsx

```diff
@@ -0,0 +1,232 @@
+            setCaretPosition(inputRef.current?.selectionEnd || 0);
+        }
+
+        if (inputRef.current) {
+            inputRef.current.addEventListener('change', updateCaretPosition);
+            inputRef.current.addEventListener('keypress', updateCaretPosition);
+            inputRef.current.addEventListener('keyup', updateCaretPosition);
+            inputRef.current.addEventListener('mousedown', updateCaretPosition);
+            inputRef.current.addEventListener('touchstart', updateCaretPosition);
+            inputRef.current.addEventListener('input', updateCaretPosition);
+            inputRef.current.addEventListener('paste', updateCaretPosition);
+            inputRef.current.addEventListener('cut', updateCaretPosition);
+            inputRef.current.addEventListener('mousemove', updateCaretPosition);
+            inputRef.current.addEventListener('select', updateCaretPosition);
+            inputRef.current.addEventListener('selectstart', updateCaretPosition);
+        }
+
+        return () => {
+            if (inputRef.current) {
+                inputRef.current.removeEventListener('change', updateCaretPosition);
+                inputRef.current.removeEventListener('keypress', updateCaretPosition);
+                inputRef.current.removeEventListener('mousedown', updateCaretPosition);
+                inputRef.current.removeEventListener('keyup', updateCaretPosition);
+                inputRef.current.removeEventListener('touchstart', updateCaretPosition);
+                inputRef.current.removeEventListener('input', updateCaretPosition);
+                inputRef.current.removeEventListener('paste', updateCaretPosition);
+                inputRef.current.removeEventListener('cut', updateCaretPosition);
+                inputRef.current.removeEventListener('mousemove', updateCaretPosition);
+                inputRef.current.removeEventListener('select', updateCaretPosition);
+                inputRef.current.removeEventListener('selectstart', updateCaretPosition);
+            }
+        };
+    }, [inputRef.current]);
+
+    const [providerResults, suggestionsHeader] = useSearchSuggestions(searchType, searchTerms, caretPosition, getCaretPosition, setSelectedOption);
+
+    const handleKeyDown = useCallback((e: React.KeyboardEvent<Element>): void => {
+        if (Keyboard.isKeyPressed(e as any, KeyCodes.ESCAPE)) {
+            e.stopPropagation();
+            e.preventDefault();
+            if (!providerResults || providerResults?.items.length === 0 || selectedOption === -1) {
+                onClose();
+            } else {
+                setSelectedOption(-1);
+            }
+        }
+
+        if (Keyboard.isKeyPressed(e as any, KeyCodes.DOWN)) {
+            e.stopPropagation();
+            e.preventDefault();
+            const totalItems = providerResults?.items.length || 0;
+            if ((selectedOption + 1) < totalItems) {
+                setSelectedOption(selectedOption + 1);
```

Comment:
To keep the current behavior, no, it shouldn't. But if @matthewbirtch thinks that is the right approach for the new interface, we can change it quickly in another PR. @matthewbirtch I think is now on holidays, but whenever he comes back, he can give his feedback here and we can add another PR.

Reply:
It's not an accessibility requirement, but is a good practice to wrap. The `DOWN` key should wrap from the last item to the first and, the `UP` key should wrap from the first to the last item. This can be in a separate PR though for sure.